packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,398 @@
+        this.artifactInfo.datasourcesDir,
+        'datasource',
+        true,
+      );
+      debug(`datasourcesList from src/datasources : ${datasourcesList}`);
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    const availableDatasources = datasourcesList.filter(item => {
+      debug(`data source unfiltered list: ${item}`);
+      const result = this._isConnectorOfType(
+        VALID_CONNECTORS_FOR_REPOSITORY,
+        item,
+      );
+      return result;
+    });
+
+    if (availableDatasources.includes(cmdDatasourceName)) {
+      Object.assign(this.artifactInfo, {
+        dataSourceClassName: cmdDatasourceName,
+      });
+    }
+
+    debug(
+      `artifactInfo.dataSourceClassName ${
+        this.artifactInfo.dataSourceClassName
+      }`,
+    );
+
+    if (availableDatasources.length === 0) {
+      return this.exit(
+        `${ERROR_NO_DATA_SOURCES_FOUND} ${this.artifactInfo.datasourcesDir}.
+        ${chalk.yellow(
+          'Please visit http://loopback.io/doc/en/lb4/Controller-generator.html for information on how repositories are discovered',
+        )}`,
+      );
+    }
+
+    return this.prompt([
+      {
+        type: 'list',
+        name: 'dataSourceClassName',
+        message: PROMPT_MESSAGE_DATA_SOURCE,
+        choices: availableDatasources,
+        when: !this.artifactInfo.dataSourceClassName,
+        default: availableDatasources[0],
+        validate: utils.validateClassName,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        debug(`props after datasource prompt: ${inspect(props)}`);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during datasource prompt: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  async promptModels() {
+    await this._inferRepositoryType();
+
+    let modelList;
+    try {
+      debug(`model list dir ${this.artifactInfo.modelDir}`);
+      modelList = await utils.getArtifactList(
+        this.artifactInfo.modelDir,
+        'model',
+      );
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    if (this.options.model) {
+      debug(`Model name received from command line: ${this.options.model}`);
+
+      this.options.model = utils.toClassName(this.options.model);
+      // assign the model name from the command line only if it is valid
+      if (modelList.length > 0 && modelList.includes(this.options.model)) {
+        Object.assign(this.artifactInfo, {modelNameList: [this.options.model]});
+      } else {
+        modelList = [];
+      }
+    }
+
+    if (modelList.length === 0) {
+      return this.exit(
+        `${ERROR_NO_MODELS_FOUND} ${this.artifactInfo.modelDir}.
+        ${chalk.yellow(
+          'Please visit http://loopback.io/doc/en/lb4/Repository-generator.html for information on how models are discovered',
+        )}`,
+      );
+    }
+
+    return this.prompt([
+      {
+        type: 'checkbox',
```

Comment:
I am little bit confused about this - why would we allow the developer to pick multiple models for a single repository? What use case do you have in mind?

I would expect that a single repository class is tied to a single model class. (Relations are out of scope of this pull request, we don't know yet whether they will use the same `lb4 relation` command or perhaps a different command specific to relations.)

Let's discuss!

Reply:
Oh, I see.

I would not choose this UX design myself, but I am fine to follow the opinion of the rest of the team here.

Reply:
> I would not choose this UX design myself

in theory this PR hide this detail from the app developer, in that he/she can pass in the command line a single repository.

Reply:
I left Artifact Generator and include support for name argument to follow the standards. However, the name is only used for the first model in a multi-select model generation process.